mlflow/pipelines/steps/evaluate.py

```diff
@@ -81,8 +81,10 @@ def _check_validation_criteria(self, metrics, validation_criteria):
             metric_name = val_criterion["metric"]
             metric_val = metrics.get(metric_name)
             if metric_val is None:
-                summary[metric_name] = False
-                continue
+                raise MlflowException(
+                    f"The metric {metric_name} was not returned from mlflow evaluation.",
```

Comment:
Can we specify that this metric was defined as part of the validation criteria for evaluation? This will add some extra context to the error